### DISCUSSION TITLE
Custom tool: Get current Date / Time

### DISCUSSION THREAD

**Discussion by yavisht** [FlowiseAI/Flowise#860] [Show and tell]
Sharing a custom tool that can grab the current day, date, time as per a specific timezone. Here's how it works with setup instructions. What custom tool would you like to make? Let me know if you plan to use this for your next project!

## The function
```
// Define the desired time zone
const timeZone = 'Australia/Sydney';

// Define an options object to format the date and time
const options = {
    timeZone: timeZone,   // Set the time zone
    year: 'numeric',       // Display the year as a number
    month: 'long',         // Display the full month name
    day: 'numeric',        // Display the day of the month as a number
    weekday: 'long',      // Display the full weekday name
    hour: '2-digit',       // Display the hour in 2-digit format (00-23)
    minute: '2-digit',     // Display the minute in 2-digit format (00-59)
    second: '2-digit',     // Display the second in 2-digit format (00-59)
    hour12: true           // Use 12-hour time format (AM/PM)
};

// Create a new Date object representing the current date and time
const today = new Date();

// Format the current date and time according to the options and locale 'en-GB'
const formattedDate = today.toLocaleString('en-GB', options);

// Create a result object containing the formatted date and the time zone
const result = {
    "formattedDate": formattedDate, // Store the formatted date and time
    "timezone": timeZone           // Store the specified time zone
};

// Return the result object as a JSON string
return JSON.stringify(result);
```

## Save the json export below and load file

<img width="619" alt="Screenshot 2023-09-01 at 9 37 40 am" src="https://github.com/FlowiseAI/Flowise/assets/6112201/7d973fad-69c3-4c9c-abf6-4b32de40fd11">

```
{
  "name": "todays_date_time",
  "description": "Useful to get todays day, date and time.",
  "color": "linear-gradient(rgb(117,118,129), rgb(230,10,250))",
  "iconSrc": "",
  "schema": "[]",
  "func": "const timeZone = 'Australia/Sydney';\nconst options = {\n    timeZone: timeZone,\n    year: 'numeric',\n    month: 'long',\n    day: 'numeric',\n    weekday: 'long',\n    hour: '2-digit',\n    minute: '2-digit',\n    second: '2-digit',\n    hour12: true\n};\nconst today = new Date();\nconst formattedDate = today.toLocaleString('en-GB', options);\nconst result = {\n    \"formattedDate\": formattedDate,\n    \"timezone\": timeZone\n};\nreturn JSON.stringify(result);\n"
}
```
## Set it up with a function agent:
<img width="597" alt="Screenshot 2023-09-01 at 9 14 10 am" src="https://github.com/FlowiseAI/Flowise/assets/6112201/d53bce84-7c45-437e-8677-8b2885b8a5e0">

## Results:
<img width="413" alt="Screenshot 2023-09-01 at 9 15 05 am" src="https://github.com/FlowiseAI/Flowise/assets/6112201/aa01f0db-4a81-4c35-bf90-87bb18432fcf">

**Comment by HenryHengZJ** [comment]
This is super awesome!! Wish more contribution like this! Do you want to open a PR for merging this tool into the marketplace section?

**Reply by yavisht** [reply]
Thanks @HenryHengZJ  I will open a PR shortly 👍

**Reply by yavisht** [reply]
@HenryHengZJ added it here -> https://github.com/FlowiseAI/Flowise/pull/861

**Comment by Makande1982** [comment]
Thanks!

**Comment by zkPurr** [comment]
Thanks so much! Awesome work

**Comment by boogiemanders** [comment]
how do i "save the json export below and load file"?

**Reply by yavisht** [reply]
@boogiemanders that json may be outdated but the tool is now available in Flowise (See screenshot below)

![image](https://github.com/user-attachments/assets/6a88c18d-5f35-4baf-9f5e-da2f5f1b54e0)

**Comment by ghuroux** [comment]
I cannot seem to get South Africa timezone i.e. GMT+2 work. South Africa/Cape Town. Any suggestions?

**Reply by yavisht** [reply]
@ghuroux  - You can find the full list here -> https://data.iana.org/time-zones/data/backward

You can use this one -> ```const timeZone = 'Africa/Johannesburg';```

<img width="401" alt="image" src="https://github.com/user-attachments/assets/d9985da7-c01d-446f-a6a4-7c911c5ef323" />
